app/src/ui/delete-branch/delete-branch-dialog.tsx

```diff
@@ -8,31 +8,71 @@ import { ButtonGroup } from '../lib/button-group'
 import { Checkbox, CheckboxValue } from '../lib/checkbox'
 import { Dialog, DialogContent, DialogFooter } from '../dialog'
 import { Ref } from '../lib/ref'
+import { checkBranchExistsOnRemote } from '../../lib/git'
 
 interface IDeleteBranchProps {
   readonly dispatcher: Dispatcher
   readonly repository: Repository
   readonly branch: Branch
-  readonly existsOnRemote: boolean
   readonly onDismissed: () => void
 }
 
 interface IDeleteBranchState {
+  readonly existsOnRemote: boolean | null
   readonly includeRemoteBranch: boolean
 }
 
 export class DeleteBranch extends React.Component<
   IDeleteBranchProps,
   IDeleteBranchState
 > {
+  private checkingRemote = false
+
   public constructor(props: IDeleteBranchProps) {
     super(props)
 
     this.state = {
+      existsOnRemote: null,
       includeRemoteBranch: false,
     }
   }
 
+  private checkCompleted = () => {
+    this.checkingRemote = false
+  }
+
+  public componentWillReceiveProps(nextProps: IDeleteBranchProps) {
+    if (this.checkingRemote) {
```

Comment:
Triggering the check on `componentWillReceiveProps` means we rely on getting multiple prop updates during the lifetime of the dialog since `componentWillReceiveProps` isn't called for the initial mount. I'm sure we're getting more than one update right now but we really don't need to given that the props we pass in are stable. We should apply the `receiveProps` pattern here instead, see https://github.com/desktop/desktop/blob/458a427b012fe4c6465852905b745fe921221c69/app/src/ui/changes/commit-message.tsx#L120-L122